bin/live/pycbc_live_single_trigger_fits

```diff
@@ -198,110 +177,193 @@ for filename in files:
-alphas = {i: np.zeros(args.num_duration_bins, dtype=np.float32) for i in args.ifos}
-counts = {i: np.zeros(args.num_duration_bins, dtype=np.float32) for i in args.ifos}
+n_bins = duration_bin_edges.size - 1
+alphas = {i: np.zeros(n_bins, dtype=np.float32) for i in args.ifos}
+counts = {i: np.zeros(n_bins, dtype=np.float32) for i in args.ifos}
+binned_sngl_stats = {ifo: {} for ifo in args.ifos}
+
 for ifo in args.ifos:
-    for key in events[ifo].data:
-        fout['{}/triggers/{}'.format(ifo, key)] = events[ifo].data[key]
-    event_bin = np.array([tbins[d] for d in events[ifo].data['template_duration']])
-    for bin_num in range(args.num_duration_bins):
+    # Sort the events into their bins
+    event_bin = np.array([tbins[d]
+                          for d in events[ifo].data['template_duration']])
+
+    # For each bin, do the fit
+    for bin_num in range(n_bins):
         inbin = event_bin == bin_num
+        if not np.count_nonzero(inbin):
+            # No triggers, alpha and count are -1
+            counts[ifo][bin_num] = -1
+            alphas[ifo][bin_num] = -1
+            continue
+
+        # For ease, keep the sngl_ranking values as 
+        binned_sngl_stats[ifo][bin_num] = \
+            events[ifo].data[args.sngl_ranking][inbin]
+
         counts[ifo][bin_num] = np.count_nonzero(inbin)
         alphas[ifo][bin_num], _ = trstats.fit_above_thresh(args.fit_function,
-                                      events[ifo].data[args.fit_param][inbin],
+                                      binned_sngl_stats[ifo][bin_num],
                                       args.fit_threshold)
-    fout['{}/fit_coeff'.format(ifo)] = alphas[ifo]
-    fout['{}/counts'.format(ifo)] = counts[ifo]
-    fout[ifo].attrs['live_time'] = live_time[ifo]
 
-fout['bins_upper'] = tbins.upper()
-fout['bins_lower'] = tbins.lower()
+with h5py.File(oput_file, 'w') as fout:
 
-fout.attrs['analysis_date'] = args.analysis_date
-fout.attrs['input'] = sys.argv
-fout.attrs['cuts'] = ineq_str
-fout.attrs['fit_function'] = args.fit_function
-fout.attrs['fit_threshold'] = args.fit_threshold
+    for ifo in args.ifos:
+        fout.create_group(ifo)
+        # Save the triggers we have used for the fits
+        fout[ifo].create_group('triggers')
+        for key in events[ifo].data:
+            fout[ifo]['triggers'][key] = events[ifo].data[key]
+
+        fout[ifo]['fit_coeff'] = alphas[ifo]
+        fout[ifo]['counts'] = counts[ifo]
+        fout[ifo].attrs['live_time'] = live_time[ifo]
+
+    fout['bins_upper'] = tbins.upper()
+    fout['bins_lower'] = tbins.lower()
+
+    fout.attrs['analysis_date'] = args.analysis_date
+    fout.attrs['input'] = sys.argv
+    fout.attrs['cuts'] = args.template_cuts + args.trigger_cuts
+    fout.attrs['fit_function'] = args.fit_function
+    fout.attrs['fit_threshold'] = args.fit_threshold
+
+if not args.output_fit_plot:
+    logging.info("Done")
+    exit(0)
 
-fout.close()
-logging.info("Done")
+import matplotlib
```

Comment:
Here onwards is some code I had around from doing the plots of the fits. The output is, for example, like this:

![image](https://user-images.githubusercontent.com/44500294/195366328-00742a5c-5283-4c85-b71b-f77542e01a4f.png)

Reply:
We should definitely make these nice plots, but I suggest splitting the plotting code to a separate script. We do not necessarily want to put the plots in the same place as the fit results anyway, they should probably go together with the daily lag and single trigger plots.

Reply:
For info - updated plot using the reversed rainbow
![image](https://user-images.githubusercontent.com/44500294/202752379-2f0d2438-9107-4bb6-926d-a4aa5629cc0e.png)